cylc/flow/cfgspec/workflow.py

```diff
@@ -1362,24 +1395,76 @@ def get_script_common_text(this: str, example: Optional[str] = None):
                         "[task events]submission timeout"
                     )
                 ))
-                # TODO: add descriptions for the handlers below. Some of them
-                # didn't have any mention in the Cylc 7 suiterc ref docs, but
-                # a look at git blame indicates none of them are new in Cylc 8.
-                Conf('expired handlers', VDR.V_STRING_LIST, None)
-                Conf('late offset', VDR.V_INTERVAL, None)
-                Conf('late handlers', VDR.V_STRING_LIST, None)
-                Conf('submitted handlers', VDR.V_STRING_LIST, None)
-                Conf('started handlers', VDR.V_STRING_LIST, None)
-                Conf('succeeded handlers', VDR.V_STRING_LIST, None)
-                Conf('failed handlers', VDR.V_STRING_LIST, None)
-                Conf('submission failed handlers', VDR.V_STRING_LIST, None)
-                Conf('warning handlers', VDR.V_STRING_LIST, None)
-                Conf('critical handlers', VDR.V_STRING_LIST, None)
-                Conf('retry handlers', VDR.V_STRING_LIST, None)
-                Conf('submission retry handlers', VDR.V_STRING_LIST, None)
-                Conf('execution timeout handlers', VDR.V_STRING_LIST, None)
-                Conf('submission timeout handlers', VDR.V_STRING_LIST, None)
-                Conf('custom handlers', VDR.V_STRING_LIST, None)
+                Conf('expired handlers', VDR.V_STRING_LIST, None, desc='''
+                    Invoke if this task has expired.
+
+                    .. caution::
+
+                       Due to changes to the Cylc 8 scheduling algorithm
+                       this event is unlikely to occur.
+                ''')
+                Conf('late offset', VDR.V_INTERVAL, None, desc='''
+                    How long to wait for this task to be triggered before
+                    running
+                    :cylc:conf:`flow.cylc[runtime][<namespace>][events]
+                    late handlers`.
+                ''')
+                Conf('late handlers', VDR.V_STRING_LIST, None, desc='''
+                    Invoke if this task is never active and is late.
+
+                    .. caution::
+
+                       Due to changes to the Cylc 8 scheduling algorithm
+                       this event is unlikely to occur until the task is about
+                       to submit anyway.
+                ''')
+                Conf('submitted handlers', VDR.V_STRING_LIST, None, desc='''
+                    Invoke if this task is submitted.
```

Comment:
```suggestion
                    Handlers to invoke if this task is submitted.
```
(I'd make this tweak for each item below)